runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java

```diff
@@ -40,204 +46,237 @@ public LogUnitClient getClient(long epoch, UUID clusterID) {
+     * Handle a read log response from the server.
      *
-     * @param msg Incoming Message
-     * @param ctx Context
-     * @param r   Router
-     * @throws OverwriteException Throws OverwriteException if address has already been written to.
+     * @param msg The read log response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return {@link ReadResponse} sent back from server.
      */
-    @ClientHandler(type = CorfuMsgType.ERROR_OVERWRITE)
-    private static Object handleOverwrite(CorfuPayloadMsg<Integer> msg, ChannelHandlerContext ctx, IClientRouter r)
-            throws Exception {
-        throw new OverwriteException(OverwriteCause.fromId(msg.getPayload()));
+    @ResponseHandler(type = PayloadCase.READ_LOG_RESPONSE)
+    private static Object handleReadLogResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return getReadResponse(msg.getPayload().getReadLogResponse());
     }
 
     /**
-     * Handle an ERROR_OOS message.
+     * Handle a inspect addresses response from the server.
      *
-     * @param msg Incoming Message
-     * @param ctx Context
-     * @param r   Router
-     * @throws OutOfSpaceException Throws OutOfSpaceException if log unit out of space.
+     * @param msg The inspect addresses response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return {@link InspectAddressesResponse} sent back from server.
      */
-    @ClientHandler(type = CorfuMsgType.ERROR_OOS)
-    private static Object handleOos(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
-            throws Exception {
-        throw new OutOfSpaceException();
+    @ResponseHandler(type = PayloadCase.INSPECT_ADDRESSES_RESPONSE)
+    private static Object handleInspectResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return getInspectAddressesResponse(msg.getPayload().getInspectAddressesResponse());
     }
 
     /**
-     * Handle an ERROR_RANK message.
+     * Handle a trim log response from the server.
      *
-     * @param msg Incoming Message
-     * @param ctx Context
-     * @param r   Router
-     * @throws Exception Throws Exception if write has been outranked.
+     * @param msg The trim log response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return Always True, since the trim log was successful.
      */
-    @ClientHandler(type = CorfuMsgType.ERROR_RANK)
-    private static Object handleOutranked(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
-            throws Exception {
-        throw new Exception("rank");
+    @ResponseHandler(type = PayloadCase.TRIM_LOG_RESPONSE)
+    private static Object handleTrimLogResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return true;
     }
```

Comment:
Can't the handler be of return type `void`?

Reply:
No, they are supposed to match the FunctionalInterface specified in `ClientResponseHandler`